### PR TITLE
ui: Added toggle Switch to hide Headers

### LINKED ISSUE #1099: [UI] Implement toggle to show/hide header fields
**Current Behavior**

Right now, all the header fields are visible on the performance page which is not looking good on UI.

**Desired Behavior**

Implement a toggle to show/hide the header fields.

**Screenshot**
Screenshot 2020-06-03 at 4 56 56 AM

---

### PR DESCRIPTION
**Pull request by dhruv0000**
Signed-off-by: Dhruv Patel <patel.4@iitj.ac.in>

**Description**
Added the switch button to toggle between labels and placeholders

This PR fixes #1099 

**Notes to reviewers**

#### Checklist

- [x] [Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin).
- [x] PR title includes `[<name of area>]:<description>`

### CODE REVIEW

**Comment by kushthedude** [comment]
Congrats @dhruv0000 on your first PR 🎉

File: ui/components/MesheryPerformanceComponent.js [hunk 1]
```diff
@@ -54,6 +54,10 @@ const styles = (theme) => ({
     marginTop: theme.spacing(3),
     marginLeft: theme.spacing(1),
   },
+  expansionPanel: {
+    boxShadow:'none',
+    border: '1px solid rgb(196,196,196)',
```
Comment (kushthedude) [inline ui/components/MesheryPerformanceComponent.js:59]: Can we remove borders from the expansion panel ?

File: ui/components/MesheryPerformanceComponent.js [hunk 2]
```diff
@@ -613,6 +558,77 @@ class MesheryPerformanceComponent extends React.Component {
                   </RadioGroup>
                 </FormControl>
               </Grid>
+              <Grid item xs={12} sm={12} gutterBottom>
+                <ExpansionPanel className={classes.expansionPanel}>
+                  <ExpansionPanelSummary expanded={true}>
+                    <Typography align="center" color="textSecondary" varient="h6">More Options</Typography>
```
Comment (kushthedude) [inline ui/components/MesheryPerformanceComponent.js:564]: Also, can we remove the border structure from the More Options & add an arrow-down/up icon on the basis of the state of the expansion panel?
Screenshot 2020-06-19 at 7 59 17 PM

Somewhat like this with an arrow icon adjacent to it.
Screenshot 2020-06-19 at 8 02 40 PM

Comment (kushthedude) [inline ui/components/MesheryPerformanceComponent.js:564]: Can we rename `More Options` to `Add Request Data` ?

**Comment by dhruv0000** [comment]
image
How about now?

**Comment by kushthedude** [comment]
Looks great @dhruv0000, but I think the previous bordered design over the options was better. Can you please make it bordered again rather than the shadowed box. It will be good to merge after that.

**Comment by dhruv0000** [comment]
image
@kushthedude

**Comment by Nikhil-Ladha** [comment]
Just a question here @kushthedude , @dhruv0000 and @leecalcote , instead of creating a button of the same width as other fields won't it be better to make a smaller one with a color to do the same, as it looks more like a drop-down to me rather than a toggler. What say?

**Comment by kushthedude** [comment]
> Just a question here @kushthedude , @dhruv0000 and @leecalcote , instead of creating a button of the same width as other fields won't it be better to make a smaller one with a color to do the same, as it looks more like a drop-down to me rather than a toggler. What say?

It is not a dropdown but an expansionPanel which should be of the same width for even UI if we make a dropdown of smaller width which then will reveal the fields of more width than the controller dropdown it will look ugly I think, a dropdown should only the fields with the width same or less then the parent IMO.
What you say @Nikhil-Ladha ?

**Comment by Nikhil-Ladha** [comment]
I just checked out the PR in my local, and it is looking pretty good. Much different than what I was thinking about. 
Another thing, the advanced option isn't appealing or doesn't looks like an expansion panel and more like any other field, just a 'font-weight:bold' may help, like the image I have attached. Otherwise, it is all good.
Screenshot from 2020-06-20 18-28-44

**Comment by dhruv0000** [comment]
> I just checked out the PR in my local, and it is looking pretty good. Much different than what I was thinking about.
> Another thing, the advanced option isn't appealing or doesn't looks like an expansion panel and more like any other field, just a 'font-weight:bold' may help, like the image I have attached. Otherwise, it is all good.
> Screenshot from 2020-06-20 18-28-44

Hey @Nikhil-Ladha, won't making it bold emphasizes more on advanced option than other fields, and as for giving a subtle hint about the expansion, I tried using [a shadow effect](https://github.com/layer5io/meshery/pull/1151#issuecomment-646845911), but as @kushthedude later correctly advised that it was looking a bit off theme. 
Mabey something could be done in that direction?

**Comment by Nikhil-Ladha** [comment]
Well, It didn't affected much and seemed to be just as I hoped. But, if that may happen then we not consider that option.
Not sure about the shadowed one though. Current one looks good.

**Comment by kushthedude** [comment]
I tried your suggestion @Nikhil-Ladha, it was highlighting the expansion-panel heading from the other fields which should not be the case. Coming to the point of letting the user know of the drop-down the arrow icon added in the field is sufficient for now.
If needed we can iterate more into this, as we also need to make changes in our performance UI.

**Comment by kushthedude** [comment]
@leecalcote I think we are good to go with this, please have a look!

**Comment by leecalcote** [comment]
@dhruv0000 ooooh, niiice!

**Comment by leecalcote** [comment]
@dhruv0000 I can see perhaps adding a light shade of gray or some other color to signify that the Advanced Options dropdown is a group of fields. 

Untitled

Other than this, it looks fantastic! You're hired. 😉

**Pull request merged**